src/olympia/addons/forms.py

```diff
@@ -150,6 +152,17 @@ def __init__(self, *args, **kw):
     def clean_slug(self):
         return clean_addon_slug(self.cleaned_data['slug'], self.instance)
 
+    def clean_contributions(self):
+        if not self.cleaned_data['contributions']:
+            return self.cleaned_data['contributions']
+        hostname = urlsplit(self.cleaned_data['contributions']).hostname
+        for domain in amo.VALID_CONTRIBUTION_DOMAINS:
```

Comment:
`hostname.endswith(amo.VALID_CONTRIBUTION_DOMAINS)` should work, no need for a loop

Reply:
https://github.com/mozilla/addons-server/pull/6252/commits/92831dad900a2a2937b41a05b3e06c3c73c86881